### PR TITLE
Add unit tests on target/host

### LINKED ISSUE #142: Unit testing framework on host and target

### PR DESCRIPTION
**Pull request by sorru94**
Closes #142

### CODE REVIEW

File: .github/workflows/unit.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+name: Unit tests
+
+on:
+    pull_request:
+    push:
+
+jobs:
+  unit-tests-on_host:
+    runs-on: ubuntu-latest
+    container: espressif/idf:release-v5.0
```
Comment (harlem88) [inline .github/workflows/unit.yaml:30]:
```suggestion
    container: espressif/idf:release-v5.0.2
```

**Review by harlem88 — CHANGES_REQUESTED** [review]
Add  `Check clang formatting`  on `unit_tests` folder.

File: .github/workflows/unit.yaml (hunk 1 above)
Comment (harlem88) [reply .github/workflows/unit.yaml]: sorry, this is `espressif/idf:v5.0.2`

Comment (harlem88) [reply .github/workflows/unit.yaml]: use `idf:release-v5.0`.

**Comment by harlem88** [comment]
About folder naming, what do you think about renaming to :
- `unit_tests` -> `tests`
- `host_and_target` -> `generic`, `all_targets`
- `host_prj` -> `host_app`

**Comment by sorru94** [comment]
> About folder naming, what do you think about renaming to :
> 
>     * `unit_tests` -> `tests`
> 
>     * `host_and_target` -> `generic`, `all_targets`
> 
>     * `host_prj` -> `host_app`

The choice of the name `unit_tests` was mostly because I was hoping to have another folder in the future for end-to-end tests, but it's probably too early for that.
'generic' looks good and 'host_app' also.

**Comment by harlem88** [comment]
> > About folder naming, what do you think about renaming to :
> > ```
> > * `unit_tests` -> `tests`
> > 
> > * `host_and_target` -> `generic`, `all_targets`
> > 
> > * `host_prj` -> `host_app`
> > ```
> 
> The choice of the name `unit_tests` was mostly because I was hoping to have another folder in the future for end-to-end tests, but it's probably too early for that. 'generic' looks good and 'host_app' also.

Looking in the different folders of  [esp-idf/components](https://github.com/espressif/esp-idf/tree/master/components/efuse)  you can always find a couple of `test_host` and `test_apps`, perhaps this should also be a potential folder structure.

**Review by harlem88 — APPROVED** [review]

**Pull request merged**